tools/generation_planner.py

```
@@ -0,0 +1,265 @@
+          3. Query Pattern Oracle for advisory recommendations.
+          4. Score all candidate routes.
+          5. Select the best-scoring route.
+          6. Return a structured GenerationPlan.
+        """
+        # Step 1: Validate
+        GenerationIntent.create(intent, name, self.workspace)
+
+        # Step 2: Load routes
+        routes = load_generation_routes(self.workspace)
+        if not routes:
+            raise ValueError(f"no generation routes defined in workspace manifest")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)